src/app/br_discover.cpp

```diff
@@ -29,16 +29,19 @@
 #include "br_discover.hpp"
 
 #include <chrono>
+#include <sys/socket.h>
 #include <thread>
 
 #include "common/error_macros.hpp"
 #include "common/utils.hpp"
 
+#define SO_BINDTODEVICE 25
+
```

Comment:
I don't think it's acceptable to refine the macro in system headers. If you are struggling with support on MacOS, you should use `IPV6_BOUND_IF`.
```
#ifdef __linux__
    rval = setsockopt(socket, SOL_SOCKET, SO_BINDTODEVICE, aNetIf, strlen(aNetIf));
#else  // __NetBSD__ || __FreeBSD__ || __APPLE__
    rval = setsockopt(socket, IPPROTO_IPV6, IP_BOUND_IF, &aNetIfIndex, sizeof(aNetIfIndex));
#endif // __linux__
    VerifyOrDie(rval == 0, OT_EXIT_ERROR_ERRNO);
```

You need to include `#include <netinet/in.h>` for Mac OS.

I didn't capture this in https://github.com/openthread/ot-commissioner/pull/260, but please fix that one as well.

Reply:
Thanks for you suggestion. but in the job logs from the previous checks such as `java-bind` and clang-x-xxx`, I can capture the following error. 

```
/Users/runner/work/ot-commissioner/ot-commissioner/src/app/br_discover.cpp:56:59: fatal error: use of undeclared identifier 'SO_BINDTODEVICE'
    if (!aNetIf.empty() && setsockopt(socket, SOL_SOCKET, SO_BINDTODEVICE, &aNetIf[0], sizeof(aNetIf)) < 0)
                                                          ^
1 error generated.
```
actually, this micro has been defined in <sys/socket.h> (https://linux.die.net/man/7/socket). so this error not just for  Mac OS.

Reply:
Do you have a link to the failed GitHub Actions runs?

Reply:
Here is the log for `java-binding` https://github.com/openthread/ot-commissioner/actions/runs/8814617371/job/24230558759.

Reply:
`java-binding` tests are running on macOS. `clang-x-xxx` are cancelled because `java-binding` failed. So all your problems are macOS support

Reply:
Got it, let me change the code according your suggestion and have a new try.

Reply:
As you said, this failure is related to macOS support, and all tests have passed with your suggestion. Thanks.

Reply:
fixed in #263, thanks.